### PR TITLE
Add CI build for integration with plone.app.contenttypes

### PR DESCRIPTION
**Pull request by hvelarde**
Even as collective.cover has been in production using plone.app.contenttypes for more than 2 years in many sites, we haven't deal with testing that in CI.

Many changes were made here to achieve compatibility and fix some errors:

- testing content creation was refactored to simplify it
- Dexterity Image content was removed as is no longer necessary (test case is covered by using plone.app.contenttypes)
- a bunch of hacks to deal with Archetypes/Dexterity API incompatibilities were included
- RF collection tile test was refactored and finally fixed (I hope so)
- documentation was updated to include a known good versions of packages involved

This is a huge step in the search of Plone 5 compatibility.

### CODE REVIEW

**Comment by hvelarde** [comment]
@fredvd seems p.a.contenttypes Image type doesn't have a ``setImage`` method, so we'll need to refactor the tests setup.

anyway, let's ask @pbauer

**Comment by pbauer** [comment]
DX only has a image attribute and also does not accept a image but `obj.image = NamedBlobImage(data=raw_data, filename='my_image.png')`. ther is also no option to refresh exif (yet).

**Comment by hvelarde** [comment]
this is now a blocker for #579

File: src/collective/cover/tiles/file.py [hunk 1]
```diff
@@ -79,10 +79,19 @@ def download_widget(self):
         obj = uuidToObject(self.data['uuid'])
         if obj:
             url = obj.absolute_url()
-            icon = obj.getBestIcon()
             portal_url = obj.portal_url()
-            mime = obj.lookupMime(obj.getField('file').getContentType(obj))
-            size = obj.get_size()
+
+            try:
```
Comment (hvelarde) [inline src/collective/cover/tiles/file.py]: @pbauer @thet this is the kind of things I'm stuck with; I need your wisdom on the best way to implement this and/or fix plone.app.contenttypes

**Comment by hvelarde** [comment]
@idgserpro do you mind help me testing this branch?

**Comment by hvelarde** [comment]
@rodfersou please review and merge.

**Comment by davilima6** [comment]
@hvelarde, no need for changelog in this one?

**Comment by hvelarde** [comment]
mostly changes on test infrastructure because of incompatibilities among Archetypes and Dexterity and some bug fixes; but, yes, you are probably right.

**Pull request merged**